src/Logic.hs

```diff
@@ -311,6 +313,10 @@ handlePullRequestClosed pr state = Pr.deletePullRequest pr <$>
         pure state { Pr.integrationCandidate = Nothing }
       _notCandidatePr -> pure state
 
+handlePullRequestEdited :: PullRequestId -> Text -> ProjectState -> Action ProjectState
+handlePullRequestEdited prId newTitle = pure . Pr.updatePullRequest prId changeTitleAndBaseBranch
+  where changeTitleAndBaseBranch pr = pr { Pr.title = newTitle }
```

Comment:
This name does not yet apply, but it is also not future proof. Maybe rename to `updatePr`?